plugins/content/caption/base/batch/CopyCaptions/KAsyncCopyCaptions.class.php

```diff
@@ -189,29 +216,124 @@ private function getCaptionContent($captionAssetId)
 
 	/**
 	 * @param KalturaCopyCaptionsJobData $data
-	 * @param KalturaClipDescriptionArray $clipDescriptionArray
 	 * @throws kApplicativeException
 	 */
-	private function copyFromClipToDestination(KalturaCopyCaptionsJobData $data, $clipDescriptionArray)
+	protected function copyFromSingleSourceToDestination(KalturaCopyCaptionsJobData $data)
 	{
 		$errorMsg = '';
-		//currently only one source
-		$originalCaptionAssets = $this->getAllCaptionAsset($clipDescriptionArray[0]->sourceEntryId);
+		$clipDescriptionArray = $data->clipsDescriptionArray;
+		$originalCaptionAssets = $this->listCaptionAssets($clipDescriptionArray[0]->sourceEntryId);

```

Comment:
for sure this is not empty?

Reply:
no, based on the flow it could be empty, added condition